alf/nest/nest.py

```diff
@@ -282,3 +282,66 @@ def find_field(nest, name, ignore_empty=True):
             elif isinstance(elem, (dict, tuple, list)):
                 ret = ret + find_field(elem, name)
     return ret
+
+
+def prune_nest_like(nest, slim_nest, value_to_match=None):
+    """Prune a nested structure referring to another slim nest. Generally, for
+    every corrsponding node, we only keep the fields that're contained in
+    `slim_nest`. In addition, if a field of `slim_nest` contains a value of
+    `value_to_match`, then the corresponding field of `nest` will also be updated
+    to this value.
+
+    Note: If a node is a list or unnamedtuple, then we require their lengths are
+    equal.
+
+    For example:
+        x = dict(a=1, b=2)
+        y = dict(a=[])
```

Comment:
This is not a good example because x.a is not nested, but y.a is nested. Line 230 will fail.

Reply:
Fixed